Different "path"-dict-key-names on localized systems

Hi there, 

I'm on a german arch system. I get a key-error when I run the freshly cloned project.

```
mkdir: Verzeichnis 'obj/' angelegt
cc -flto -Wall -Wextra -Wpedantic -std=gnu11 -nostartfiles -fno-PIC -Os -fno-plt -fno-stack-protector -fno-stack-check -fno-unwind-tables -fno-asynchronous-unwind-tables -fomit-frame-pointer -ffast-math -no-pie -fno-pic -fno-PIE -m64 -march=core2 -ffunction-sections -fdata-sections -fno-plt  -m64 -I/usr/include/SDL2 -D_REENTRANT  -c "test/hello.c" -o "obj/hello.lto.o"
cc -flto -Wall -Wextra -Wpedantic -std=gnu11 -nostartfiles -fno-PIC -Os -fno-plt -fno-stack-protector -fno-stack-check -fno-unwind-tables -fno-asynchronous-unwind-tables -fomit-frame-pointer -ffast-math -no-pie -fno-pic -fno-PIE -m64 -march=core2 -ffunction-sections -fdata-sections -fno-plt  -m64 -I/usr/include/SDL2 -D_REENTRANT  -c "rt/crt1.c" -o "obj/crt1.lto.o"
cc -m64 -nostartfiles -nostdlib -r -o "obj/hello.start.o" obj/hello.lto.o obj/crt1.lto.o
python3 src/smol.py  -lc -lSDL2 -lX11  "obj/hello.start.o" "obj/symbols.hello.start.asm"
Traceback (most recent call last):
  File "src/smol.py", line 78, in <module>
    main()
  File "src/smol.py", line 61, in main
    spaths = args.libdir + paths['libraries']
KeyError: 'libraries'
make: *** [Makefile:77: obj/symbols.hello.start.asm] Fehler 1
```

I figured out, this lines, produced the errors:

```
src/smol.py:    spaths = args.libdir + paths['libraries']
src/smol.py:    libraries=paths['libraries']
```

So I printed out the paths-dict-keys on my system and found out, the keynames are localized too... (I have the key "Bibliotheken", what's the german word for "libraries", if Ichange the script to 'Bibliotheken' everything just works right)

I'm not a python-wizard, so I would come up with a fix, if I could, but I think, other people might be on a localized system too... (I tried it with a german Ubuntu on a virtual system, there are no problems there.)

--- a/README.md
+++ b/README.md
@@ -4,6 +4,13 @@ Shoddy minsize-oriented linker
 
 PoC by Shiz, bugfixing and 64-bit version by PoroCYon.
 
+## Dependencies
+
+* A C compiler (preferably GCC), GNU ld, binutils, GNU make, ...
+* nasm
+* scanelf from pax-utils
+* Python 3
+
 ## Usage
 
 ```sh

--- a/src/smolparse.py
+++ b/src/smolparse.py
@@ -71,15 +71,30 @@ def get_needed_syms(readelf_bin, inpfiles):
 
     return syms#, needgot
 
+def format_cc_path_line(entry):
+    category, path = entry.split(': ', 1)
+    path = path.lstrip('=')
+    return (category, list(set(os.path.realpath(p) \
+        for p in path.split(':') if os.path.isdir(p))))
+
 def get_cc_paths(cc_bin):
+    bak = os.environ.copy()
+    os.environ['LANG'] = "C" # DON'T output localized search dirs!
     output = subprocess.check_output([cc_bin, '-print-search-dirs'],
                                      stderr=subprocess.DEVNULL)
-    paths = {}
-    for entry in output.decode('utf-8').splitlines():
-        category, path = entry.split(': ', 1)
-        path = path.lstrip('=')
-        paths[category] = list(set(os.path.realpath(p) \
-                for p in path.split(':') if os.path.isdir(p)))
+    os.environ = bak
+
+    outputpairs = list(map(format_cc_path_line,
+                           output.decode('utf-8').splitlines()))
+    for category, path in outputpairs: paths[category] = path
+
+    if 'libraries' not in paths: # probably localized... sigh
+        # monkeypatch, assuming order...
+        paths = {}
+        paths['install'  ] = outputpairs[0][1]
+        paths['programs' ] = outputpairs[1][1]
+        paths['libraries'] = outputpairs[2][1]
+
     return paths
 
 def is_valid_elf(f):

--- a/src/smolparse.py
+++ b/src/smolparse.py
@@ -86,6 +86,8 @@ def get_cc_paths(cc_bin):
 
     outputpairs = list(map(format_cc_path_line,
                            output.decode('utf-8').splitlines()))
+    paths = {}
+
     for category, path in outputpairs: paths[category] = path
 
     if 'libraries' not in paths: # probably localized... sigh

